webrtc/RTCConfiguration-iceCandidatePoolSize.html

```diff
@@ -1,35 +1,87 @@
+-->
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
-</head>
-<body>
 <script>
-function testValidPoolSize(size, expectedSize)
-{
-    test(function() {
-        var pc = new RTCPeerConnection({iceCandidatePoolSize: size})
-        assert_equals(pc.getConfiguration().iceCandidatePoolSize, expectedSize !== undefined ? expectedSize : size)
-    }, "Setting iceCandidatePoolSize to a valid value: " + size)
-}
-
-function testInvalidPoolSize(size)
-{
-    test(function() {
-        assert_throws(new TypeError(), () => new RTCPeerConnection({iceCandidatePoolSize: size}))
-    }, "Setting iceCandidatePoolSize to an invalid value: " + size)
-}
-
-testValidPoolSize(10)
-testValidPoolSize(0)
-testValidPoolSize(null, 0)
-testValidPoolSize(undefined, 0)
-
-testInvalidPoolSize(-1)
-testInvalidPoolSize(1000)
-testInvalidPoolSize("string")
+  function initializeWithValidPoolSize(iceCandidatePoolSize, expectedSize) {
+    test(() => {
+      const pc = new RTCPeerConnection({
+        iceCandidatePoolSize
+      });
+      assert_equals(pc.getConfiguration().iceCandidatePoolSize, expectedSize);
+
+    }, `new RTCPeerConnection({ iceCandidatePoolSize }) with valid value: ${iceCandidatePoolSize}`);
+  }
+
+  function initializeWithInvalidPoolSize(iceCandidatePoolSize) {
+    test(() => {
+      assert_throws(new TypeError(), () => {
+        new RTCPeerConnection({
+          iceCandidatePoolSize
+        });
+      });
+    }, `new RTCPeerConnection({ iceCandidatePoolSize }) with invalid value: ${iceCandidatePoolSize}`);
+  }
+
+  function reconfigureWithValidPoolSize(iceCandidatePoolSize) {
+    test(() => {
+      const pc = new RTCPeerConnection();
+      pc.setConfiguration({ iceCandidatePoolSize });
+      assert_equals(pc.getConfiguration().iceCandidatePoolSize, expectedSize);
+
+    }, `setConfiguration({ iceCandidatePoolSize }) to a valid value: ${iceCandidatePoolSize}`);
+  }
+
+  function reconfigureWithInvalidPoolSize(iceCandidatePoolSize) {
+    test(() => {
+      const pc = new RTCPeerConnection();
+      assert_throws(new TypeError(), () => {
+        pc.setConfiguration({ iceCandidatePoolSize });
+      });
+    }, `setConfiguration({ iceCandidatePoolSize }) to an invalid value: ${iceCandidatePoolSize}`);
+  }
+
+  /*
+
+  dictionary RTCConfiguration {
+      ...
+      [EnforceRange]
+      octet                    iceCandidatePoolSize = 0;
+  };
+
+  ... of type octet
+  */
+  initializeWithValidPoolSize(0, 0);
+  initializeWithValidPoolSize(127, 127);
+  initializeWithValidPoolSize(255, 255);
+
+  // Defaulting to 0
+  initializeWithValidPoolSize(null, 0);
+  initializeWithValidPoolSize(undefined, 0);
+
+  // [EnforceRange]
+  initializeWithInvalidPoolSize(-1);
+  initializeWithInvalidPoolSize(256);
+
+  /*
+  setConfiguration({ iceCandidatePoolSize })
+  */
+  reconfigureWithValidPoolSize(0, 0);
```

Comment:
(copied from below)

I would actually prefer to avoid introducing additional testing mechanism logic whenever possible. TBH, I would prefer if the function abstractions didn't exist at all, and that the entirety of this file was structured similar to these examples: 

- https://github.com/w3c/web-platform-tests/commit/7f89c4349a940ad78d87a86e6a6bbd6ca5415921
- https://github.com/w3c/web-platform-tests/commit/79092e1bb755102e9e387dfa89ce676794226b1a
- https://github.com/w3c/web-platform-tests/commit/044ae0edd0da2320150b7b6edbdb1a49e9a026a0
- https://github.com/w3c/web-platform-tests/commit/d94b87c3b1ba302502155c0e364cee11c4eec393
- https://github.com/w3c/web-platform-tests/commit/348334c42edf5d77028c765412d3958f5d1ceffb